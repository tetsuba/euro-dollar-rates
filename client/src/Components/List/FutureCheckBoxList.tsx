import React from 'react'
import List from './List'
import Checkbox from '../Form/Checkbox'
import { EventClickType, FutureType } from '../../utils/types'

interface FutureListTypes {
    month: string
    year: string
    price: string
    checked: boolean
}

interface PropTypes {
    list: FutureType[]
    onClick: (e: EventClickType) => void
}

export default function FutureCheckBoxList({
    list,
    onClick,
}: PropTypes): JSX.Element {
    function renderLinkItem(
        { month, year, price, checked }: FutureListTypes,
        index: number
    ) {
        return (
            <Checkbox
                key={`cb-${index}`}
                index={index}
                label={`${year}/${month} - ${price}`}
                checked={checked}
                onClick={onClick}
            />
        )
    }
    return <List renderItem={renderLinkItem} list={list} partialKey="fcbl-" />
}
