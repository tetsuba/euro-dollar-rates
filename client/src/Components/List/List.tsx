import React from 'react'
import ListHeader from './ListHeader'

export interface PropTypes {
    list: any[]
    renderItem: (args: any, i: number) => JSX.Element
    listHeader?: string
    partialKey: string
}

export default function List(props: PropTypes): JSX.Element {
    const { list, renderItem, listHeader, partialKey } = props

    function renderListItem(args: any, i: number) {
        return (
            <li className="list-group-item" key={`${partialKey + i}`}>
                {renderItem(args, i)}
            </li>
        )
    }

    return (
        <ul className="list-group list-group-flush">
            {listHeader && (
                <ListHeader>
                    <h5 className="mt-3">{listHeader}</h5>
                </ListHeader>
            )}
            {list.map(renderListItem)}
        </ul>
    )
}
