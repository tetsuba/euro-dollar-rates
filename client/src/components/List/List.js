import React from 'react'
import { Link } from 'react-router-dom'

export default function List(props) {
    const { list, label, link } = props

    function getListItem(name) {
        return (
            <li className="list-group-item" key={name}>
                {link ? (
                    <a href={`${link}${name}`} target="_blank" rel="noreferrer">
                        {name}
                    </a>
                ) : (
                    <Link to={`/market-watch/futures/${name}`}>{name}</Link>
                )}
            </li>
        )
    }

    return (
        <ul className="list-group list-group-flush">
            <li className="list-group-item shadow-sm">
                <h5 className="mt-3">{label}</h5>
            </li>
            {list.map(getListItem)}
        </ul>
    )
}
