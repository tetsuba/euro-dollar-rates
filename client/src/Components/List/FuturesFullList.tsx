import React from 'react'
import List from './List'

interface PropTypes {
    list: string[]
    text: string
}

export default function FuturesFullList({ list, text }: PropTypes): React.ReactElement {
    const renderList = text !== ''

    function renderLinkItem(name: string) {
        return (
            <a
                href={`https://www.marketwatch.com/investing/future/${name}`}
                rel="noopener"
                data-testid="external-link"
            >
                {name}
            </a>
        )
    }

    return renderList
        ? (
            <List
                partialKey="ffl-"
                renderItem={renderLinkItem}
                listHeader="Full List"
                list={list}
            />
        )
        : <></>
}
