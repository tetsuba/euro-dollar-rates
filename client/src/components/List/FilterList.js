import React from 'react'
import List from './List';

export default function FilterList(props) {
  const {list, text, label, link} = props
  return (
    (text !== '')
    && <List
        link={link}
        label={label}
        list={list.filter((name) => name.startsWith(text))}
      />
  )
}