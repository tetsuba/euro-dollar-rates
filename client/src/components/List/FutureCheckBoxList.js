import React from 'react'
import List from './List'
import Checkbox from '../Form/Checkbox'

export default function FutureCheckBoxList({ list, onClick }) {
    function renderLinkItem({ month, year, price, checked }, index) {
        return (
            <Checkbox
                key={`cb-${index}`}
                index={index}
                label={`${year}/${month} - ${price}`}
                checked={checked}
                onClick={onClick}
            />
        )
    }

    return <List renderItem={renderLinkItem} list={list} partialKey="fcbl-" />
}
