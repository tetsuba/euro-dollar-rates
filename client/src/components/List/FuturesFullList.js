import React from 'react'
import List from './List'

export default function FuturesFullList({ list, text }) {
    function renderLinkItem(name) {
        return (
            <a
                href={`https://www.marketwatch.com/investing/future/${name}`}
                rel="noopener"
                data-testid="external-link"
            >
                {name}
            </a>
        )
    }

    return (
        text !== '' && (
            <List
                partialKey="ffl-"
                renderItem={renderLinkItem}
                listHeader="Full List"
                list={list.filter((name) => name.startsWith(text))}
            />
        )
    )
}
