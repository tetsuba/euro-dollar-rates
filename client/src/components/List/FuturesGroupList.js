import React from 'react'
import { Link } from 'react-router-dom'
import List from './List'

export default function FuturesGroupList({list, text}) {
  function renderLinkItem(name) {
    return (
      <Link
        to={`/market-watch/futures/${name}`}
        data-testid="internal-link"
      >
        {name}
      </Link>
    )
  }

  return (
    text !== '' && (
      <List
        partialKey="fgl-"
        renderItem={renderLinkItem}
        listHeader="Group List"
        list={list.filter((name) => name.startsWith(text))}
      />
    )
  )
}
