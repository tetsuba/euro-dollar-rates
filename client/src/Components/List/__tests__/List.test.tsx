import React from 'react'
import { render } from '@testing-library/react'
import List, { PropTypes } from '../List'

describe('List', () => {
    const mockProps: PropTypes = {
        list: ['mock item 1', 'mock item 2'],
        renderItem: (name) => <div>{name}</div>,
        partialKey: 'mock-',
    }

    test('render with list header ', () => {
        const { asFragment } = render(
            <List {...mockProps} listHeader="Mock Header" />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test('render without list header ', () => {
        const { asFragment } = render(<List {...mockProps} />)
        expect(asFragment()).toMatchSnapshot()
    })
})
