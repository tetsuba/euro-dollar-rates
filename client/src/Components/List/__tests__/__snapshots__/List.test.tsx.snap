// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render with list header  1`] = `
<DocumentFragment>
  <ul
    class="list-group list-group-flush"
  >
    <li
      class="list-group-item shadow-sm"
    >
      <h5
        class="mt-3"
      >
        Mock Header
      </h5>
    </li>
    <li
      class="list-group-item"
    >
      <div>
        mock item 1
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div>
        mock item 2
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`List render without list header  1`] = `
<DocumentFragment>
  <ul
    class="list-group list-group-flush"
  >
    <li
      class="list-group-item"
    >
      <div>
        mock item 1
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div>
        mock item 2
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
